torch/testing/_internal/common_methods_invocations.py

```diff
@@ -325,6 +325,10 @@ def _getattr_qual(obj, name, default=_NOTHING):
 #   Each SampleInput defines an "input", "args", "kwargs",
 #   an "output_process_fn_grad" function, the "broadcasts_input" bool and
 #   a "name".
+#   All the "sample_inputs" functions are invoked within a `torch.no_grad()`
+#   environment for efficiency and correctness. As such remember to set the the
+#   "requires_grad" flag on the inputs **after** performing any transformations
```

Comment:
What's a little scary about this change is that we have no way of validating if the things that should require grad still do after this, right? Like that evaluation is just done manually?

Reply:
It is true that, if no input from an opinfo has `requires_grad=True`, then we throw, but it could be that there are some operations that return just one of the inputs with `requries_grad` but not all of them.

One way to check this would be to throw if for an input of floating point, none of the inputs has `requires_grad=True`. This might give a few false positives, but I think it might even uncover some previous errors we missed.